RedfishPkg/Include/Library/RedfishHttpLib.h

```diff
@@ -0,0 +1,328 @@
+
+**/
+EFI_STATUS
+RedfishCleanupService (
+  IN  REDFISH_SERVICE  RedfishService
+  );
+
+/**
+  This function returns JSON value in given RedfishPayload. Returned JSON value
+  is a reference to the JSON value in RedfishPayload. Any modification to returned
+  JSON value will change JSON value in RedfishPayload.
+
+  @param[in]  This            Pointer to EDKII_REDFISH_HTTP_PROTOCOL instance.
+  @param[in]  RedfishPayload  Pointer to Redfish payload.
+
+  @retval     EDKII_JSON_VALUE   JSON value is returned.
+  @retval     NULL               Errors occur.
+
+**/
+EDKII_JSON_VALUE
+RedfishJsonInPayload (
+  IN  REDFISH_PAYLOAD  RedfishPayload
+  );
+
+/**
+  This function free resources in Request. Request is no longer available
+  after this function returns successfully.
+
+  @param[in]  Request      HTTP request to be released.
+
+  @retval     EFI_SUCCESS     Resrouce is released successfully.
+  @retval     Others          Errors occur.
+
+**/
+EFI_STATUS
+RedfishHttpFreeRequest (
+  IN  REDFISH_REQUEST  *Request
+  );
+
+/**
+  This function free resources in Response. Response is no longer available
+  after this function returns successfully.
+
+  @param[in]  Response     HTTP response to be released.
+
+  @retval     EFI_SUCCESS     Resrouce is released successfully.
+  @retval     Others          Errors occur.
+
+**/
+EFI_STATUS
+RedfishHttpFreeResponse (
+  IN  REDFISH_RESPONSE  *Response
+  );
+
+/**
+  This function expire the cached response of given URI.
+
+  @param[in]  Uri          Target response of URI.
+
+  @retval     EFI_SUCCESS     Target response is expired successfully.
+  @retval     Others          Errors occur.
+
+**/
+EFI_STATUS
+RedfishHttpExpireResponse (
+  IN  EFI_STRING  Uri
+  );
+
+/**
+  Get redfish resource from given resource URI with cache mechanism
+  supported. It's caller's responsibility to Response by calling
+  RedfishHttpFreeResponse ().
+
+  @param[in]  Service       Redfish service instance to make query.
+  @param[in]  Uri           Target resource URI.
+  @param[in]  Request       Additional request context. This is optional.
+  @param[out] Response      HTTP response from redfish service.
+  @param[in]  UseCache      If it is TRUE, this function will search for
+                            cache first. If it is FALSE, this function
+                            will query Redfish URI directly.
+
+  @retval     EFI_SUCCESS     Resrouce is returned successfully.
+  @retval     Others          Errors occur.
+
+**/
+EFI_STATUS
+RedfishHttpGetResource (
+  IN  REDFISH_SERVICE   Service,
+  IN  EFI_STRING        Uri,
+  IN  REDFISH_REQUEST   *Request OPTIONAL,
+  OUT REDFISH_RESPONSE  *Response,
+  IN  BOOLEAN           UseCache
+  );
+
+/**
+  Perform HTTP PATCH to send redfish resource to given resource URI.
+  It's caller's responsibility to free Response by calling RedfishHttpFreeResponse ().
+
+  @param[in]  Service       Redfish service instance to make query.

```

Comment:
Is the comment "Redfish service instance to make query." accurate? Or "Redfish service instance to perform HTTP PATCH"?